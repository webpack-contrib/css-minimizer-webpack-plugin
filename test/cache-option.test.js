import path from "path";

import del from "del";

import CssMinimizerPlugin from "../src/index";

import {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  readAssets,
} from "./helpers";

describe('"cache" option', () => {
  const fileSystemCacheDirectory = path.resolve(
    __dirname,
    "./outputs/type-filesystem"
  );
  const fileSystemCacheDirectory1 = path.resolve(
    __dirname,
    "./outputs/type-filesystem-1"
  );
  const fileSystemCacheDirectory2 = path.resolve(
    __dirname,
    "./outputs/type-filesystem-2"
  );

  beforeAll(() =>
    Promise.all([
      del(fileSystemCacheDirectory),
      del(fileSystemCacheDirectory1),
      del(fileSystemCacheDirectory2),
    ])
  );

  it('should work with the "false" value for the "cache" option', async () => {
    const compiler = getCompiler({
      entry: {
        one: `${__dirname}/fixtures/cache.js`,
        two: `${__dirname}/fixtures/cache-1.js`,
        three: `${__dirname}/fixtures/cache-2.js`,
        four: `${__dirname}/fixtures/cache-3.js`,
        five: `${__dirname}/fixtures/cache-4.js`,
      },
      cache: false,
    });

    new CssMinimizerPlugin().apply(compiler);

    let getCounter = 0;

    compiler.cache.hooks.get.tap(
      { name: "TestCache", stage: -100 },
      (identifier) => {
        if (identifier.indexOf("CssMinimizerWebpackPlugin") !== -1) {
          getCounter += 1;
        }
      }
    );

    let storeCounter = 0;

    compiler.cache.hooks.store.tap(
      { name: "TestCache", stage: -100 },
      (identifier) => {
        if (identifier.indexOf("CssMinimizerWebpackPlugin") !== -1) {
          storeCounter += 1;
        }
      }
    );

    const stats = await compile(compiler);

    // Without cache webpack always try to get
    expect(getCounter).toBe(5);
    // Without cache webpack always try to store
    expect(storeCounter).toBe(5);
    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");

    getCounter = 0;
    storeCounter = 0;

    const newStats = await compile(compiler);

    // Without cache webpack always try to get
    expect(getCounter).toBe(5);
    // Without cache webpack always try to store
    expect(storeCounter).toBe(5);
    expect(readAssets(compiler, newStats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(newStats)).toMatchSnapshot("errors");
    expect(getWarnings(newStats)).toMatchSnapshot("warnings");

    await new Promise((resolve) => {
      compiler.close(() => {
        resolve();
      });
    });
  });

  it('should work with the "memory" value for the "cache.type" option', async () => {
    const compiler = getCompiler({
      entry: {
        one: `${__dirname}/fixtures/cache.js`,
        two: `${__dirname}/fixtures/cache-1.js`,
        three: `${__dirname}/fixtures/cache-2.js`,
        four: `${__dirname}/fixtures/cache-3.js`,
        five: `${__dirname}/fixtures/cache-4.js`,
      },
      cache: {
        type: "memory",
      },
    });

    new CssMinimizerPlugin().apply(compiler);

    let getCounter = 0;

    compiler.cache.hooks.get.tap(
      { name: "TestCache", stage: -100 },
      (identifier) => {
        if (identifier.indexOf("CssMinimizerWebpackPlugin") !== -1) {
          getCounter += 1;
        }
      }
    );

    let storeCounter = 0;

    compiler.cache.hooks.store.tap(
      { name: "TestCache", stage: -100 },
      (identifier) => {
        if (identifier.indexOf("CssMinimizerWebpackPlugin") !== -1) {
          storeCounter += 1;
        }
      }
    );

    const stats = await compile(compiler);

    // Get cache for assets
    expect(getCounter).toBe(5);
    // Store cached assets
    expect(storeCounter).toBe(5);
    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");

    getCounter = 0;
    storeCounter = 0;

    const newStats = await compile(compiler);

    // Get cache for assets
    expect(getCounter).toBe(5);
    // No need to store, we got cached assets
    expect(storeCounter).toBe(0);
    expect(readAssets(compiler, newStats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(newStats)).toMatchSnapshot("errors");
    expect(getWarnings(newStats)).toMatchSnapshot("warnings");

    await new Promise((resolve) => {
      compiler.close(() => {
        resolve();
      });
    });
  });

  it('should work with the "filesystem" value for the "cache.type" option', async () => {
    const compiler = getCompiler({
      entry: {
        one: `${__dirname}/fixtures/cache.js`,
        two: `${__dirname}/fixtures/cache-1.js`,
        three: `${__dirname}/fixtures/cache-2.js`,
        four: `${__dirname}/fixtures/cache-3.js`,
        five: `${__dirname}/fixtures/cache-4.js`,
      },
      cache: {
        type: "filesystem",
        cacheDirectory: fileSystemCacheDirectory,
      },
    });

    new CssMinimizerPlugin().apply(compiler);

    let getCounter = 0;

    compiler.cache.hooks.get.tap(
      { name: "TestCache", stage: -100 },
      (identifier) => {
        if (identifier.indexOf("CssMinimizerWebpackPlugin") !== -1) {
          getCounter += 1;
        }
      }
    );

    let storeCounter = 0;

    compiler.cache.hooks.store.tap(
      { name: "TestCache", stage: -100 },
      (identifier) => {
        if (identifier.indexOf("CssMinimizerWebpackPlugin") !== -1) {
          storeCounter += 1;
        }
      }
    );

    const stats = await compile(compiler);

    // Get cache for assets
    expect(getCounter).toBe(5);
    // Store cached assets
    expect(storeCounter).toBe(5);
    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");

    getCounter = 0;
    storeCounter = 0;

    const newStats = await compile(compiler);

    // Get cache for assets
    expect(getCounter).toBe(5);
    // No need to store, we got cached assets
    expect(storeCounter).toBe(0);
    expect(readAssets(compiler, newStats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(newStats)).toMatchSnapshot("errors");
    expect(getWarnings(newStats)).toMatchSnapshot("warnings");

    await new Promise((resolve) => {
      compiler.close(() => {
        resolve();
      });
    });
  });

  it('should work with the "filesystem" value for the "cache.type" option and source maps', async () => {
    const compiler = getCompiler({
      devtool: "source-map",
      entry: {
        one: path.resolve(__dirname, "./fixtures/cache.js"),
        two: path.resolve(__dirname, "./fixtures/cache-1.js"),
        three: path.resolve(__dirname, "./fixtures/cache-2.js"),
        four: path.resolve(__dirname, "./fixtures/cache-3.js"),
        five: path.resolve(__dirname, "./fixtures/cache-4.js"),
      },
      cache: {
        type: "filesystem",
        cacheDirectory: fileSystemCacheDirectory1,
      },
    });

    new CssMinimizerPlugin().apply(compiler);

    let getCounter = 0;

    compiler.cache.hooks.get.tap(
      { name: "TestCache", stage: -100 },
      (identifier) => {
        if (identifier.indexOf("CssMinimizerWebpackPlugin") !== -1) {
          getCounter += 1;
        }
      }
    );

    let storeCounter = 0;

    compiler.cache.hooks.store.tap(
      { name: "TestCache", stage: -100 },
      (identifier) => {
        if (identifier.indexOf("CssMinimizerWebpackPlugin") !== -1) {
          storeCounter += 1;
        }
      }
    );

    const stats = await compile(compiler);

    // Get cache for assets
    expect(getCounter).toBe(5);
    // Store cached assets
    expect(storeCounter).toBe(5);
    expect(readAssets(compiler, stats, /\.css(\.map)?$/)).toMatchSnapshot(
      "assets"
    );
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");

    getCounter = 0;
    storeCounter = 0;

    const newStats = await compile(compiler);

    // Get cache for assets
    expect(getCounter).toBe(5);
    // No need to store, we got cached assets
    expect(storeCounter).toBe(0);
    expect(readAssets(compiler, newStats, /\.css(\.map)?$/)).toMatchSnapshot(
      "assets"
    );
    expect(getErrors(newStats)).toMatchSnapshot("errors");
    expect(getWarnings(newStats)).toMatchSnapshot("warnings");

    await new Promise((resolve) => {
      compiler.close(() => {
        resolve();
      });
    });
  });

  it('should work with the "filesystem" value for the "cache.type" option and output warnings', async () => {
    const compiler = getCompiler({
      entry: {
        one: path.resolve(__dirname, "./fixtures/cache.js"),
        two: path.resolve(__dirname, "./fixtures/cache-1.js"),
        three: path.resolve(__dirname, "./fixtures/cache-2.js"),
        four: path.resolve(__dirname, "./fixtures/cache-3.js"),
        five: path.resolve(__dirname, "./fixtures/cache-4.js"),
      },
      cache: {
        type: "filesystem",
        cacheDirectory: fileSystemCacheDirectory2,
      },
    });

    new CssMinimizerPlugin({
      minify: (data) => {
        // eslint-disable-next-line global-require
        const postcss = require("postcss");
        const [[fileName, input]] = Object.entries(data);
        const plugin = () => {
          return {
            postcssPlugin: "warning-plugin",
            OnceExit(decl, { result }) {
              result.warn(`Warning from ${result.opts.from}`, {
                plugin: "warning-plugin",
              });
            },
          };
        };

        plugin.postcss = true;

        return postcss([plugin])
          .process(input, { from: fileName, to: fileName })
          .then((result) => {
            return {
              code: result.css,
              map: result.map,
              error: result.error,
              warnings: result.warnings(),
            };
          });
      },
    }).apply(compiler);

    let getCounter = 0;

    compiler.cache.hooks.get.tap(
      { name: "TestCache", stage: -100 },
      (identifier) => {
        if (identifier.indexOf("CssMinimizerWebpackPlugin") !== -1) {
          getCounter += 1;
        }
      }
    );

    let storeCounter = 0;

    compiler.cache.hooks.store.tap(
      { name: "TestCache", stage: -100 },
      (identifier) => {
        if (identifier.indexOf("CssMinimizerWebpackPlugin") !== -1) {
          storeCounter += 1;
        }
      }
    );

    const stats = await compile(compiler);

    // Get cache for assets
    expect(getCounter).toBe(5);
    // Store cached assets
    expect(storeCounter).toBe(5);
    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");

    getCounter = 0;
    storeCounter = 0;

    const newStats = await compile(compiler);

    // Get cache for assets
    expect(getCounter).toBe(5);
    // No need to store, we got cached assets
    expect(storeCounter).toBe(0);
    expect(readAssets(compiler, newStats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(newStats)).toMatchSnapshot("errors");
    expect(getWarnings(newStats)).toMatchSnapshot("warnings");

    await new Promise((resolve) => {
      compiler.close(() => {
        resolve();
      });
    });
  });
});
