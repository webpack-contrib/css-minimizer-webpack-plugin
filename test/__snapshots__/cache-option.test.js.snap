// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"cache" option should work with the "false" value for the "cache" option: assets 1`] = `
Object {
  "five.css": "body{color:red;font-size:20px}a{color:blue}",
  "four.css": "h1{color:green}h2{color:#ff0}",
  "one.css": "a{text-align:center}",
  "three.css": "body{color:red;font-size:20px}a{color:blue}",
  "two.css": "a{text-align:center}",
}
`;

exports[`"cache" option should work with the "false" value for the "cache" option: assets 2`] = `
Object {
  "five.css": "body{color:red;font-size:20px}a{color:blue}",
  "four.css": "h1{color:green}h2{color:#ff0}",
  "one.css": "a{text-align:center}",
  "three.css": "body{color:red;font-size:20px}a{color:blue}",
  "two.css": "a{text-align:center}",
}
`;

exports[`"cache" option should work with the "false" value for the "cache" option: errors 1`] = `Array []`;

exports[`"cache" option should work with the "false" value for the "cache" option: errors 2`] = `Array []`;

exports[`"cache" option should work with the "false" value for the "cache" option: warnings 1`] = `Array []`;

exports[`"cache" option should work with the "false" value for the "cache" option: warnings 2`] = `Array []`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and output warnings: assets 1`] = `
Object {
  "five.css": "body {
  color: red;
}
body {
  color: red;
  font-size: 20px;
}
a {
  color: blue;
}

",
  "four.css": "h1 {
  color: green;
}
h2 {
  color: yellow;
}
h3 {
}

",
  "one.css": "a {
  text-align: center;
}

",
  "three.css": "body {
  color: red;
}
body {
  color: red;
  font-size: 20px;
}
a {
  color: blue;
}

",
  "two.css": "a {
  text-align: center;
}

",
}
`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and output warnings: assets 2`] = `
Object {
  "five.css": "body {
  color: red;
}
body {
  color: red;
  font-size: 20px;
}
a {
  color: blue;
}

",
  "four.css": "h1 {
  color: green;
}
h2 {
  color: yellow;
}
h3 {
}

",
  "one.css": "a {
  text-align: center;
}

",
  "three.css": "body {
  color: red;
}
body {
  color: red;
  font-size: 20px;
}
a {
  color: blue;
}

",
  "two.css": "a {
  text-align: center;
}

",
}
`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and output warnings: errors 1`] = `Array []`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and output warnings: errors 2`] = `Array []`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and output warnings: warnings 1`] = `
Array [
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from five.css ",
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from four.css ",
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from one.css ",
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from three.css ",
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from two.css ",
]
`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and output warnings: warnings 2`] = `
Array [
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from five.css ",
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from four.css ",
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from one.css ",
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from three.css ",
  "Warning: Css Minimizer Plugin: warning-plugin: Warning from two.css ",
]
`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and source maps: assets 1`] = `
Object {
  "five.css": "body{color:red;font-size:20px}a{color:blue}
/*# sourceMappingURL=five.css.map*/",
  "five.css.map": "{\\"version\\":3,\\"file\\":\\"five.css\\",\\"mappings\\":\\"AAGA,KACE,SAAU,CACV,cACF,CACA,EACE,UACF\\",\\"sources\\":[\\"webpack:///./test/bar1.css\\"],\\"sourcesContent\\":[\\"body {\\\\n  color: red;\\\\n}\\\\nbody {\\\\n  color: red;\\\\n  font-size: 20px;\\\\n}\\\\na {\\\\n  color: blue;\\\\n}\\\\n\\"],\\"names\\":[],\\"sourceRoot\\":\\"\\"}",
  "four.css": "h1{color:green}h2{color:#ff0}
/*# sourceMappingURL=four.css.map*/",
  "four.css.map": "{\\"version\\":3,\\"file\\":\\"four.css\\",\\"mappings\\":\\"AAAA,GACE,WACF,CACA,GACE,UACF\\",\\"sources\\":[\\"webpack:///./test/bar2.css\\"],\\"sourcesContent\\":[\\"h1 {\\\\n  color: green;\\\\n}\\\\nh2 {\\\\n  color: yellow;\\\\n}\\\\nh3 {\\\\n}\\\\n\\"],\\"names\\":[],\\"sourceRoot\\":\\"\\"}",
  "one.css": "a{text-align:center}
/*# sourceMappingURL=one.css.map*/",
  "one.css.map": "{\\"version\\":3,\\"file\\":\\"one.css\\",\\"mappings\\":\\"AAAA,EACE,iBACF\\",\\"sources\\":[\\"webpack:///./test/foo.css\\"],\\"sourcesContent\\":[\\"a {\\\\n  text-align: center;\\\\n}\\\\n\\"],\\"names\\":[],\\"sourceRoot\\":\\"\\"}",
  "three.css": "body{color:red;font-size:20px}a{color:blue}
/*# sourceMappingURL=three.css.map*/",
  "three.css.map": "{\\"version\\":3,\\"file\\":\\"three.css\\",\\"mappings\\":\\"AAGA,KACE,SAAU,CACV,cACF,CACA,EACE,UACF\\",\\"sources\\":[\\"webpack:///./test/bar1.css\\"],\\"sourcesContent\\":[\\"body {\\\\n  color: red;\\\\n}\\\\nbody {\\\\n  color: red;\\\\n  font-size: 20px;\\\\n}\\\\na {\\\\n  color: blue;\\\\n}\\\\n\\"],\\"names\\":[],\\"sourceRoot\\":\\"\\"}",
  "two.css": "a{text-align:center}
/*# sourceMappingURL=two.css.map*/",
  "two.css.map": "{\\"version\\":3,\\"file\\":\\"two.css\\",\\"mappings\\":\\"AAAA,EACE,iBACF\\",\\"sources\\":[\\"webpack:///./test/foo.css\\"],\\"sourcesContent\\":[\\"a {\\\\n  text-align: center;\\\\n}\\\\n\\"],\\"names\\":[],\\"sourceRoot\\":\\"\\"}",
}
`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and source maps: assets 2`] = `
Object {
  "five.css": "body{color:red;font-size:20px}a{color:blue}
/*# sourceMappingURL=five.css.map*/",
  "five.css.map": "{\\"version\\":3,\\"file\\":\\"five.css\\",\\"mappings\\":\\"AAGA,KACE,SAAU,CACV,cACF,CACA,EACE,UACF\\",\\"sources\\":[\\"webpack:///./test/bar1.css\\"],\\"sourcesContent\\":[\\"body {\\\\n  color: red;\\\\n}\\\\nbody {\\\\n  color: red;\\\\n  font-size: 20px;\\\\n}\\\\na {\\\\n  color: blue;\\\\n}\\\\n\\"],\\"names\\":[],\\"sourceRoot\\":\\"\\"}",
  "four.css": "h1{color:green}h2{color:#ff0}
/*# sourceMappingURL=four.css.map*/",
  "four.css.map": "{\\"version\\":3,\\"file\\":\\"four.css\\",\\"mappings\\":\\"AAAA,GACE,WACF,CACA,GACE,UACF\\",\\"sources\\":[\\"webpack:///./test/bar2.css\\"],\\"sourcesContent\\":[\\"h1 {\\\\n  color: green;\\\\n}\\\\nh2 {\\\\n  color: yellow;\\\\n}\\\\nh3 {\\\\n}\\\\n\\"],\\"names\\":[],\\"sourceRoot\\":\\"\\"}",
  "one.css": "a{text-align:center}
/*# sourceMappingURL=one.css.map*/",
  "one.css.map": "{\\"version\\":3,\\"file\\":\\"one.css\\",\\"mappings\\":\\"AAAA,EACE,iBACF\\",\\"sources\\":[\\"webpack:///./test/foo.css\\"],\\"sourcesContent\\":[\\"a {\\\\n  text-align: center;\\\\n}\\\\n\\"],\\"names\\":[],\\"sourceRoot\\":\\"\\"}",
  "three.css": "body{color:red;font-size:20px}a{color:blue}
/*# sourceMappingURL=three.css.map*/",
  "three.css.map": "{\\"version\\":3,\\"file\\":\\"three.css\\",\\"mappings\\":\\"AAGA,KACE,SAAU,CACV,cACF,CACA,EACE,UACF\\",\\"sources\\":[\\"webpack:///./test/bar1.css\\"],\\"sourcesContent\\":[\\"body {\\\\n  color: red;\\\\n}\\\\nbody {\\\\n  color: red;\\\\n  font-size: 20px;\\\\n}\\\\na {\\\\n  color: blue;\\\\n}\\\\n\\"],\\"names\\":[],\\"sourceRoot\\":\\"\\"}",
  "two.css": "a{text-align:center}
/*# sourceMappingURL=two.css.map*/",
  "two.css.map": "{\\"version\\":3,\\"file\\":\\"two.css\\",\\"mappings\\":\\"AAAA,EACE,iBACF\\",\\"sources\\":[\\"webpack:///./test/foo.css\\"],\\"sourcesContent\\":[\\"a {\\\\n  text-align: center;\\\\n}\\\\n\\"],\\"names\\":[],\\"sourceRoot\\":\\"\\"}",
}
`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and source maps: errors 1`] = `Array []`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and source maps: errors 2`] = `Array []`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and source maps: warnings 1`] = `Array []`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option and source maps: warnings 2`] = `Array []`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option: assets 1`] = `
Object {
  "five.css": "body{color:red;font-size:20px}a{color:blue}",
  "four.css": "h1{color:green}h2{color:#ff0}",
  "one.css": "a{text-align:center}",
  "three.css": "body{color:red;font-size:20px}a{color:blue}",
  "two.css": "a{text-align:center}",
}
`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option: assets 2`] = `
Object {
  "five.css": "body{color:red;font-size:20px}a{color:blue}",
  "four.css": "h1{color:green}h2{color:#ff0}",
  "one.css": "a{text-align:center}",
  "three.css": "body{color:red;font-size:20px}a{color:blue}",
  "two.css": "a{text-align:center}",
}
`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option: errors 1`] = `Array []`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option: errors 2`] = `Array []`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option: warnings 1`] = `Array []`;

exports[`"cache" option should work with the "filesystem" value for the "cache.type" option: warnings 2`] = `Array []`;

exports[`"cache" option should work with the "memory" value for the "cache.type" option: assets 1`] = `
Object {
  "five.css": "body{color:red;font-size:20px}a{color:blue}",
  "four.css": "h1{color:green}h2{color:#ff0}",
  "one.css": "a{text-align:center}",
  "three.css": "body{color:red;font-size:20px}a{color:blue}",
  "two.css": "a{text-align:center}",
}
`;

exports[`"cache" option should work with the "memory" value for the "cache.type" option: assets 2`] = `
Object {
  "five.css": "body{color:red;font-size:20px}a{color:blue}",
  "four.css": "h1{color:green}h2{color:#ff0}",
  "one.css": "a{text-align:center}",
  "three.css": "body{color:red;font-size:20px}a{color:blue}",
  "two.css": "a{text-align:center}",
}
`;

exports[`"cache" option should work with the "memory" value for the "cache.type" option: errors 1`] = `Array []`;

exports[`"cache" option should work with the "memory" value for the "cache.type" option: errors 2`] = `Array []`;

exports[`"cache" option should work with the "memory" value for the "cache.type" option: warnings 1`] = `Array []`;

exports[`"cache" option should work with the "memory" value for the "cache.type" option: warnings 2`] = `Array []`;
